massdriver/client.go

```diff
@@ -58,24 +63,47 @@ func NewMassdriverClient(deployment_id, token, event_topic_arn string) (*Massdri
 	return c, nil
 }
 
-func (c MassdriverClient) PublishEventToSNS(event *Event) error {
-	jsonBytes, err := json.Marshal(event)
+func (c MassdriverClient) PublishEventToSNS(event *Event, diags *diag.Diagnostics) error {
+	input, err := c.buildSNSEvent(event)
+
 	if err != nil {
+		*diags = diag.FromErr(err)
+		return err
+	}
+
+	if c.EventTopicARN != "" {
+		_, err = c.SNSClient.Publish(context.Background(), &input)
+		*diags = diag.FromErr(err)
 		return err
+	} else {
+		eventBytes, _ := json.Marshal(input)
+		fmt.Println("test")
+		*diags = append(*diags, diag.Diagnostic{
+			Severity: diag.Warning,
+			Summary:  "Development Override in effect. Resource will not be updated in Massdriver.",
+			Detail:   string(eventBytes),
+		})
+	}
+
+	return nil
+}
+
+func (c MassdriverClient) buildSNSEvent(event *Event) (sns.PublishInput, error) {
```

Comment:
```suggestion
func (c MassdriverClient) buildSNSEvent(event *Event) (*sns.PublishInput, error) {
```
This should return a pointer since its what the AWS library expects, and so we aren't copying the struct on the stack when we return it.